File: SortRoutines/OneRecord.java
import java.util.Scanner;
public class OneRecord
{
/*********************************************************************
 * Instance variables.
**/
  Long element;

/*********************************************************************
 * Constructor.
**/
  public OneRecord()
  {
    this.setElement(Long.MIN_VALUE);
  }

/*********************************************************************
 * Accessors.
**/
/*********************************************************************
 * Method to get the value of the 'element'.
 *
 * @return element the 'element' for this 'OneRecord'. 
**/
  public long getElement()
  {
    return this.element;
  }

/*********************************************************************
 * Mutators.
**/
/*********************************************************************
 * Method to set the value of the 'element'.
 *
 * @param value the value to be set.
**/
  public void setElement(long value)
  {
    this.element = value;
  }

/*********************************************************************
 * General methods.
**/
/*********************************************************************
 * Method to read from the input file and set the element.
 *
 * @param inFile the 'Scanner' input file from which to read.
**/
  public void readElement(Scanner inFile)
  {
    long inValue;
    inValue = inFile.nextInt();
    this.setElement(inValue);
  }

/*********************************************************************
 * Method to compare the element against another value.
 *
 * @param secondElt the other value to compare against.
**/
  public long compare(long secondElt)
  {
    long returnValue;

    returnValue = 0;
    if(this.getElement() < secondElt)
      returnValue = -1;
    else if(this.getElement() > secondElt)
      returnValue = 1;

    return returnValue;
  }

/*********************************************************************
 * Method to return a 'String' of the 'element'. 
 *
 * @return the 'String'.
**/
  public String toString()
  {
    String s = String.format("%8d",this.getElement());

    return s;
  }

} // public class OneRecord

File: mydirectory/sect3group1_hw08/OneRecord.cpp
#include "OneRecord.h"
/*****************************************************************************
 * Class for a single record
 * A record simply has a long long integer value.
 *
 * Author: Nicholas Wrobel
 *         John Quinn
 *
 * Date: 7 December 2014
**/

/*****************************************************************************
 * Constructor and destructor
**/
OneRecord::OneRecord()
{
  this->setElement(DUMMYVALUE);
}

OneRecord::~OneRecord()
{
}

/*****************************************************************************
 * Accessors and mutators
**/
long long OneRecord::getElement() const
{
  return this->element;
}

void OneRecord::setElement(long long value)
{
  this->element = value;
}

/*****************************************************************************
 * General functions
**/

/*****************************************************************************
 * Function 'readElement'
 * This function takes in a scanner and reads the next integer value, storing
 * it as the element
 *
 * Parameters
 *    inFile: the scanner to be read from
**/
void OneRecord::readElement(Scanner& inFile)
{
  long long inValue = 0;
  inValue = inFile.nextInt();
  this->setElement(inValue);
}

/*****************************************************************************
 * Function 'toString'
 * This function returns a string of the instance data (the element)
 *
 * Returns
 *    string of the instance data
**/
string OneRecord::toString() const
{
  stringstream sstm; 
  sstm << this->getElement();

  return sstm.str();
}
